src/core/qgsvectorlayerutils.cpp

```diff
@@ -901,3 +905,58 @@ QgsAttributeMap QgsVectorLayerUtils::QgsFeatureData::attributes() const
 {
   return mAttributes;
 }
+
+QHash<QString, QSet<QgsSymbolLayerId>> QgsVectorLayerUtils::labelMasks( const QgsVectorLayer *layer )
+{
+  QHash<QString, QSet<QgsSymbolLayerId>> refs;
+  // collect masked symbol layers from labeling
+  if ( layer->labelsEnabled() &&
+       layer->labeling()->settings().format().mask().enabled() )
```

Comment:
Good point. I have no clue ... I need to test

Reply:
Well, it does not work with rule based labels as you have already guessed here :-/

I am wandering if something using your StyleEntity visitor would not be better. I think it works for label settings. This could be extended also to collect symbol layers. And would also get rid of using the deprecated QgsLegendSymbolItem::originalSymbol(). What do you think ?

Reply:
I've did that and it seems a bit cleaner